algorithms/unit_tests/TestStdAlgorithmsSearch_n.cpp

```diff
@@ -261,45 +261,45 @@ void run_single_scenario(const InfoType& scenario_info, std::size_t count,
 template <class Tag, class ValueType>
 void run_all_scenarios() {
   using scenario_t = std::tuple<std::string, std::size_t, std::string>;
-  std::vector<scenario_t> scenarios;
-  scenarios.push_back(std::make_tuple("empty", 0u, "none"));
-  scenarios.push_back(std::make_tuple("one-element-a", 1u, "none"));
-  scenarios.push_back(std::make_tuple("one-element-b", 1u, "left"));
-
-  scenarios.push_back(std::make_tuple("two-elements-a", 2u, "none"));
-  scenarios.push_back(std::make_tuple("two-elements-b", 2u, "left"));
-  scenarios.push_back(std::make_tuple("two-elements-c", 2u, "right"));
-
-  scenarios.push_back(std::make_tuple("three-elements-a", 3u, "none"));
-  scenarios.push_back(std::make_tuple("three-elements-b", 3u, "left"));
-  scenarios.push_back(std::make_tuple("three-elements-c", 3u, "random"));
-  scenarios.push_back(std::make_tuple("three-elements-d", 3u, "right"));
-
-  scenarios.push_back(std::make_tuple("four-elements-a", 4u, "none"));
-  scenarios.push_back(std::make_tuple("four-elements-b", 4u, "left"));
-  scenarios.push_back(std::make_tuple("four-elements-c", 4u, "random"));
-  scenarios.push_back(std::make_tuple("four-elements-d", 4u, "right"));
-
-  scenarios.push_back(std::make_tuple("small-a", 13u, "none"));
-  scenarios.push_back(std::make_tuple("small-b", 13u, "left"));
-  scenarios.push_back(std::make_tuple("small-c", 13u, "random"));
-  scenarios.push_back(std::make_tuple("small-d", 13u, "right"));
-  scenarios.push_back(std::make_tuple("small-e", 131u, "none"));
-  scenarios.push_back(std::make_tuple("small-f", 131u, "left"));
-  scenarios.push_back(std::make_tuple("small-g", 131u, "random"));
-  scenarios.push_back(std::make_tuple("small-h", 131u, "right"));
-
-  scenarios.push_back(std::make_tuple("medium-a", 21103u, "none"));
-  scenarios.push_back(std::make_tuple("medium-b", 21103u, "left"));
-  scenarios.push_back(std::make_tuple("medium-c", 21103u, "random"));
-  scenarios.push_back(std::make_tuple("medium-d", 21103u, "right"));
-  scenarios.push_back(std::make_tuple("medium-e", 21103u, "left_and_1567"));
-  scenarios.push_back(std::make_tuple("medium-f", 21103u, "11133_and_right"));
-
-  scenarios.push_back(std::make_tuple("large-a", 101333u, "none"));
-  scenarios.push_back(std::make_tuple("large-b", 101333u, "left"));
-  scenarios.push_back(std::make_tuple("large-c", 101333u, "random"));
-  scenarios.push_back(std::make_tuple("large-d", 101333u, "right"));
+  std::vector<scenario_t> scenarios(31);
```

Comment:
I know it is just a test (so not on a critical path), but why not something simple like:

```
scenarios_t senarios[] = {
  {"empty", 0u, "none"},
  {"one-element-a", 1u, "none"},
  // ...
};

```

This works because:
The size is known at compile time, so an array is preferred.
`make_tuple` is only needed for deduction, which isn't needed in this case as you know the exact type of the tuple.
`tuple` will work fine with brace initialization syntax.

Reply:
@nliber thanks, i can make those changes!

Reply:
@nliber change made!

Reply:
changed it back after discussing with @nliber